server/src/server.ts

```diff
@@ -295,5 +314,214 @@ function getDiagnostic(info: any, map: any): Diagnostic {
+                return makeLocation(pathData.jsonMap.pointers['/buffers/' + result]);
+            }
+            else if (part === 'index') {
+                return makeLocation(pathData.jsonMap.pointers['/textures/' + result]);
+            }
+            else if (part === 'sampler') {
+                if (inChannels) {
+                    return makeLocation(pathData.jsonMap.pointers[currentAnimationPath + '/samplers/' + result]);
+                } else {
+                    return makeLocation(pathData.jsonMap.pointers['/samplers/' + result]);
+                }
+            }
+            else if (part === 'source') {
+                return makeLocation(pathData.jsonMap.pointers['/images/' + result]);
+            }
+            else if (part === 'camera') {
+                return makeLocation(pathData.jsonMap.pointers['/cameras/' + result]);
+            } else if (part === 'fragmentShader' || part === 'vertexShader') {
+                return makeLocation(pathData.jsonMap.pointers['/shaders/' + result]);
+            }
+        }
+        else {
+            if (part === 'nodes' || part === 'children' || part === 'joints') {
+                inNodes = true;
+            }
+            else if (part === 'channels') {
+                inChannels = true;
+                currentAnimationPath = currentPath.substring(0, currentPath.length - '/channels'.length);
+            }
+            else if (result.uri !== undefined) {
+                if (!result.uri.startsWith('data:') && !currentPath.startsWith('/images/')) {
+                    return makeLocation(null, Url.resolve(textDocumentPosition.textDocument.uri, result.uri));
+                } else {
+                    let uri = 'gltf-dataUri://' + encodeURIComponent(Uri.parse(textDocumentPosition.textDocument.uri).fsPath) + currentPath;
+                    return makeLocation(null, uri);
+                }
+            } else if (part === 'accessors') {
+                inAccessors = true;
+            } else if (inAccessors && !path.includes('bufferView')) {
+                let uri = 'gltf-dataUri://' + encodeURIComponent(Uri.parse(textDocumentPosition.textDocument.uri).fsPath) + currentPath;
+                return makeLocation(null, uri);
+            }
+        }
+    }
+
+    return null
+});
+
+connection.onHover((textDocumentPosition: TextDocumentPositionParams): Hover => {
+    let pathData = getPath(textDocumentPosition);
+    if (!pathData) {
+        return null;
+    }
+
+    let path = pathData.path;
+
+    if (path.startsWith('/images/')) {
+        let imageData = getFromPath(pathData.jsonMap.data, path);
+        if (imageData) {
+            if (!path.endsWith('/uri')) {
+                imageData = imageData.uri;
+            }
+            if (!imageData.startsWith('data:')) {
+                imageData = Url.resolve(textDocumentPosition.textDocument.uri, imageData);
+            }
+            let contents: MarkedString[] = [`![${path}](${imageData})`];
```

Comment:
Would love for there to be a way to limit the embedded image size.  I don't know of any easy way to do that though, so that may just become a backlog item.

Reply:
vscode just caps it at 1MB and then won't display it... that seems a bit overkill which is why I did redid this a couple times.